debian/control

```diff
@@ -60,6 +60,8 @@ Build-Depends: cmake (>= 3.10.2),
 # Crimson      libyaml-cpp-dev,
                librabbitmq-dev,
                librdkafka-dev,
+               lua-dev
```

Comment:
please use `find_package(Lua)` for checking the existence of Lua libraries. could you please double check if the runtime dependencies related to Lua are added to `radosgw` by `${shlibs:Depends}` ?

Reply:
in the code I don't need lua (or lua-dev) since we have a lua submodule which provides header and lib that we need for compilation.
however, the luarocks binary does need the lua header and lib in the cases that it compiles C code (some lua rocks require compilation, and I added a special flag for that).

I guess that there are several options here:
1. keep the submodule, and install lua-dev for luarocks to work (what I currently did)
2. keep the submodule, and install header and lib from the submodule (currently they are not installed and just used internally) - not 100% sure on how to do that, especially in the vstart case
3. remove the submodule (need to make sure that we don't break cls_lua), and use `find_package(Lua)` instead internally, and install  lua-dev for luarocks

Reply:
I think (2) is a non-starter. We don't want to fake installation or direct compiled modules to use our Lua source. I don't have a strong opinion on (1) or (3). I would defer to @tchaikov 's preference. Or maybe he has an alternative.

Reply:
looking at the list of changes in the lua submodule: https://github.com/LuaDist/lua/compare/lua-5.1...ceph:lua-5.3-ceph
it looks like there was one actual change to the code in 2013 (fixing a memory leak). the rest of the changes are around cmake and the build environment.
latest major change is merge from the lua5.3 branch in 2015.
if we can get rid of this submodule, we should probably go with option (3)

Reply:
trying it out here: https://github.com/ceph/ceph/pull/37940